File: surf-api-bukkit/surf-api-bukkit-server/src/main/java/dev/slne/surf/surfapi/bukkit/server/libs/reflection/LibReflection.java
package dev.slne.surf.surfapi.bukkit.server.libs.reflection;

import dev.slne.surf.surfapi.core.api.reflection.SurfReflection;
import dev.slne.surf.surfapi.core.server.impl.reflection.SurfReflectionImpl;

public final class LibReflection {

  public static final PaperPluginClassLoaderProxy PAPER_PLUGIN_CLASS_LOADER_PROXY;

  static {
    final SurfReflection reflection = SurfReflectionImpl.INSTANCE;

    PAPER_PLUGIN_CLASS_LOADER_PROXY = reflection.createProxy(PaperPluginClassLoaderProxy.class);
  }
}

File: surf-api-bukkit/surf-api-bukkit-server/src/main/kotlin/dev/slne/surf/surfapi/bukkit/server/libs/reflection/LibReflection.kt
package dev.slne.surf.surfapi.bukkit.server.libs.reflection

import dev.slne.surf.surfapi.core.api.reflection.createProxy
import dev.slne.surf.surfapi.core.api.reflection.surfReflection

object LibReflection {
    @JvmField
    val PAPER_PLUGIN_CLASS_LOADER_PROXY = surfReflection.createProxy<PaperPluginClassLoaderProxy>()
}
